python_scripts/02_numerical_pipeline_scaling.py

```diff
@@ -126,29 +125,27 @@
 # data and then train a new logistic regression model on that new version of
 # the dataset.
 #
-# Let's start by checking some feature statistics or the training data.
+# Let's start by printing some statistics about the training data.
 
 # %%
 data_train.describe()
 
 # %% [markdown]
-# We see that features in the dataset have very different range. Indeed, having
-# normalized feature is important. Indeed, some algorithms make assumptions
-# regarding feature distributions and having normalized feature is usually one
-# of them.
+# We see that the dataset's features span across different ranges.
+# However, we want to have normalized features because some algorithms make
+# assumptions regarding their distribution, and being normalized is usually
+# one of them.
 #
 # ```{tip}
 # Some of the reasons for scaling features are:
 #
-# * predictor using Euclidean distance (e.g k-nearest  neighbors) should have
-#   normalized feature such that each feature contribute equally distance
-#   computation;
-# * predictor internally using gradient-descent based algorithms
-#   (e.g. logistic regression) to find optimal parameters work better
-#   and faster simplify choice of parameters as learning-rate;
-# * predictor using regularization (e.g. logistic regression) require
-#   normalized features such that the penalty is properly applied to the
-#   weights.
+# * predictors using Euclidian distance (e.g k-nearest-neighbors) should have
+#   normalized features so that each one contributes equally to the distance
+#   computation.
```

Comment:
I think that we should finish with `;` and not `.` because each item does not start with an upper case. In some way we could have written the same in a single line and the semi-column would have been used.

I would be more inclined to revert these changes everywhere :)